internal/datastore/mysql/datastore_test.go

```diff
@@ -495,6 +498,102 @@ func ChunkedGarbageCollectionTest(t *testing.T, ds datastore.Datastore) {
 	req.NoError(err)
 }
 
+func QuantizedRevisionTest(t *testing.T, b testdatastore.RunningEngineForTest) {
+	testCases := []struct {
+		testName         string
+		quantization     time.Duration
+		relativeTimes    []time.Duration
+		expectedRevision uint64
+	}{
+		{
+			"DefaultRevision",
+			1 * time.Second,
+			[]time.Duration{},
+			1,
+		},
+		{
+			"OnlyPastRevisions",
+			1 * time.Second,
+			[]time.Duration{-2 * time.Second},
+			2,
+		},
+		{
+			"OnlyFutureRevisions",
+			1 * time.Second,
+			[]time.Duration{2 * time.Second},
+			2,
+		},
+		{
+			"QuantizedLower",
+			1 * time.Second,
+			[]time.Duration{-2 * time.Second, -1 * time.Nanosecond, 0},
+			3,
+		},
+		{
+			"QuantizationDisabled",
+			1 * time.Nanosecond,
+			[]time.Duration{-2 * time.Second, -1 * time.Nanosecond, 0},
+			4,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.testName, func(t *testing.T) {
+			require := require.New(t)
+			ctx, cancel := context.WithTimeout(context.Background(), 900*time.Second)
+			defer cancel()
+
+			ds := testdatastore.RunMySQLForTesting(t, "").NewDatastore(t, func(engine, uri string) datastore.Datastore {
```

Comment:
`RunMySQLForTesting` can be pulled out of the loop, and each test just makes it's own logical datastore with `NewDatastore`

This might also help lower that 900s timeout?